helix-term/src/commands.rs

```diff
@@ -515,6 +524,90 @@ impl PartialEq for MappableCommand {
     }
 }
 
+#[derive(Clone)]
+pub struct FallbackCommand {
+    pub name: &'static str,
+    pub fun: fn(cx: &mut Context, event: KeyEvent),
+    pub with_prompt: MappableCommand,
+    pub doc: &'static str,
+}
+
+macro_rules! fallback_commands {
+    ( $($name:ident, $prompting_name:ident, $doc:literal,)* ) => {
+        $(
+            #[allow(non_upper_case_globals)]
+            pub const $name: Self = Self {
+                name: stringify!($name),
+                fun: $name,
+                with_prompt: MappableCommand::$prompting_name,
+                doc: $doc
+            };
+        )*
+
+        pub const FALLBACK_COMMAND_LIST: &'static [Self] = &[
+            $( Self::$name, )*
+        ];
+    }
+}
+impl FallbackCommand {
+    pub fn execute(&self, cx: &mut Context, event: KeyEvent) {
+        (self.fun)(cx, event)
+    }
+
+    pub fn name(&self) -> &str {
+        self.name
+    }
+
+    pub fn doc(&self) -> &str {
+        self.doc
+    }
+
+    #[rustfmt::skip]
+    fallback_commands!(
+        select_around_pair, prompt_and_select_around_pair, "Select around matching delimiter",
+        select_inside_pair, prompt_and_select_inside_pair, "Select inside matching delimiter",
+    );
+}
+
+impl fmt::Debug for FallbackCommand {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("FallbackCommand")
+            .field(&self.name())
+            .finish()
+    }
+}
+
+impl fmt::Display for FallbackCommand {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.name())
+    }
+}
+
+impl std::str::FromStr for FallbackCommand {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        FallbackCommand::FALLBACK_COMMAND_LIST
+            .iter()
+            .find(|cmd| cmd.name() == s)
+            .cloned()
+            .ok_or_else(|| anyhow!("No command named '{}'", s))
+    }
+}
+
+impl<'de> Deserialize<'de> for FallbackCommand {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        let s: &'de str = Deserialize::deserialize(deserializer)?;
+        s.parse().map_err(de::Error::custom)
+    }
+}
```

Comment:
I think we can pass on deserializing fallback commands from the config for now since that's better supported via scripting in the future.